src/app/chip_data_model.gni

```diff
@@ -288,8 +292,7 @@ template("chip_data_model") {
         ]
       } else if (cluster == "icd-management-server") {
         sources += [ "${_app_root}/clusters/${cluster}/${cluster}.cpp" ]
-
-        deps += [ "${chip_root}/src/app/icd:cluster" ]
+        public_deps += [ "${chip_root}/src/app/icd:cluster" ]
```

Comment:
I would have expected this to be a dep in all-clusters-app, esp. since the code there is a workaround that we plan to remove, instead of this being a data model public_deps thing....

Reply:
Yeah, if you do a small check, you'll find out that most includes related to ICD have the `//nogncheck` workaround. Making it a public_dep will fix that. Adding just the dep for the all-cluster-app wouldn't solve the root issue.

Reply:
`// nogncheck` was not a workaround when it was initially added. Since the headers are not always included in all builds, build cannot successfully do `gn check` unless the comment was present.

The question is, do we want all components that depend on this to have the `icd:cluster` component?

Reply:
`icd:cluster` icd:cluster is already conditionnal on the datamodel so I don't see any problem with that. If you set it in the data model you should be able to include the header files (which are already encased in a `ifdef` without having to add a nogncheck on top of it...any way a follow PR will be needed to address the ICD build system configuration once the Check-in message encryption will be used

Reply:
My actual question was: are these "public API that anyone using data_model needs" or "implementation detail that we happened to need for all-clusters-app for now"?

Reply:
For now it's mostly Implementation details that we happened to need for the all-cluster-app...

Reply:
removing that changes and moving it to the all-cluster-app